src/utils.py

```diff
@@ -0,0 +1,29 @@
+import os
+import logging
+
+def delete_empty_folders(root):
```

Comment:
instead of doing a complex walk pattern can't we just output everything into tmp or something, move it, then delete entire tmp folder?